src/hidra/receiver/asapo_transfer.py

```diff
@@ -0,0 +1,78 @@
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import argparse
+import logging
+
+from hidra import Transfer, __version__, generate_filepath
+from plugins.asapo_producer import AsapoWorker
+
+logger = logging.getLogger(__name__)
+
+
+class AsapoTransfer:
+    def __init__(self, asapo_worker, signal_host, target_host, target_port, target_dir):
+
+        self.signal_host = signal_host
+        self.targets = [[target_host, target_port, 1]]
+        self.target_dir = target_dir
+        self.asapo_worker = asapo_worker
+        self.query = Transfer("STREAM_METADATA", self.signal_host)
+        self.query.initiate(self.targets)
+
+    def start_transfer(self):
+        self.query.start()
+        self.run()
+
+    def run(self):
+        while True:
+            try:
+                [metadata, data] = self.query.get()
+                if metadata is not None and data is not None:
```

Comment:
Is `data is not None` ever true when using `"STREAM_METADATA"`?

Reply:
I don't know. I hope you know it.

Reply:
They are `None` if timeout is over.

Reply:
But under which circumstances `data` is *not* `None` and which value does it have then?

Reply:
I don't know. Line to check for None is taken from example.

Reply:
Looks like a missing deserialization step in Hidra: `data == b"null"`. Thanks for checking!
Please remove condition on `data` as I would consider that a bug.